gen3-integration-tests/tests/test_study_viewer.py

```diff
@@ -0,0 +1,158 @@
+            logger.info(f"Access revoked for user")
+
+        # Delete all the request_ids after the test is executed
+        for request_id in cls.variables["request_ids"]:
+            requestor.request_delete(request_id)
+            logger.info(f"Request {request_id} deleted")
+
+    def test_unauthorized_user_request_access(self, page):
+        """
+        Scenario: Request Access without logging in
+        Steps:
+            1. Go to StudyViewer Page
+            2. Click 'Learn More' button and see 'Login to Request Access' button
+            3. click 'Login to Request Access' button which takes you to Login Page
+            4. Login with 'user0_account' user
+            5. Go to StudyViewer Page. click on 'Learn More' button
+            6. See 'Request Access' button
+        """
+        login_page = LoginPage()
+        study_viewer = StudyViewerPage()
+        # Navigate to study viewer page before logging in
+        study_viewer.go_to(page)
+        study_viewer.click_show_details(page)
+        # See login to request access button on the dataset
+        study_viewer.click_login_request_access_button(page)
+        # Login with user0_account user
+        login_page.login(page, user="user0_account")
+        # Navigate to study viewer page
+        study_viewer.go_to(page)
+        study_viewer.click_show_details(page)
+        # Check if 'request_access' button is visible
+        request_access = page.locator(study_viewer.REQUEST_ACCESS_BUTTON)
+        request_access.wait_for()
+        screenshot(page, "RequestAccessButton")
+        assert request_access.is_visible()
+
+    def test_user_requests_access(self, page):
+        """
+        Scenario: User logs in and requests access
+        Steps:
+            1. Login with 'user0_account' user and go to StudyViewer Page
+            2. Click 'Learn More' button and click on 'Request Access' button
+            3. Update the 'request_id' to Approved State and check if you see 'Request Access' button disabled
+            4. Update the 'request_id' to Signed State and check if you see 'Request Access' button enabled
+            5. Click Download button
+        """
+        login_page = LoginPage()
+        study_viewer = StudyViewerPage()
+        requestor = Requestor()
+        # Login with user0_account user
+        login_page.go_to(page)
+        login_page.login(page, user="user0_account")
+        # Navigate to study viewer page
+        study_viewer.go_to(page)
+        study_viewer.click_show_details(page)
+        # Click on request_access button
+        study_viewer.click_request_access_button(page)
+        # Get request_id from requestor db
+        time.sleep(20)
+        request_id = requestor.get_request_id("user0_account")
+        self.variables["request_ids"].append(request_id)
+        requestor.request_approved(request_id)
+        approved_status = requestor.get_request_status(request_id)
+        if approved_status == "APPROVED":
```

Comment:
I'm not super up-to-date with all the study viewer changes, but when a request first created by the requestor, shouldn't it's initial status be `DRAFT`?
Is there some special customization for this test so it can result in `APPROVED` status?

Reply:
Initially when the request is created, yes it is in `DRAFT` status. But at L109 when are send a request to update the request status to `APPROVED` and then checking if the request_status has got updated.